network/src/transport.rs

```diff
@@ -226,7 +167,9 @@ where
 struct UpgradeContext {
     noise: NoiseUpgrader,
     handshake_version: u8,
-    own_handshake: HandshakeMsg,
+    supported_protocols: BTreeMap<MessagingProtocolVersion, SupportedProtocols>,
```

Comment:
Is this a BTreeMap over a HashMap for size performance?

Reply:
no idea, IMO we should have just used a HashMap :P

Reply:
:( yeah, okay then let's leave it.